src/parser/jobs/blm/Leylines.js

```diff
@@ -1,45 +1,117 @@
-import React from 'react'
+import React, {Fragment} from 'react'
 import ACTIONS from 'data/ACTIONS'
 import STATUSES from 'data/STATUSES'
 import {Rule, Requirement} from 'parser/core/modules/Checklist'
 import Module from 'parser/core/Module'
 import {ActionLink} from 'components/ui/DbLink'
 import {i18nMark, Trans} from '@lingui/react'
-
-const LEYLINE_DURATION = 30000
+import {Accordion, Button} from 'semantic-ui-react'
+import {Group, Item} from 'parser/core/modules/Timeline'
+import DISPLAY_ORDER from './DISPLAY_ORDER'
 
 export default class Leylines extends Module {
 	static handle = 'leylines'
 	static i18n_id = i18nMark('blm.leylines.title')
 	static title = 'Ley Lines'
+	static displayOrder = DISPLAY_ORDER.LEY_LINES
 
 	static dependencies = [
+		'timeline',
 		'checklist',
 	]
 	_circleOfPowers = {
 		current: null,
 		history: [],
 	}
-	_leyLineHistory = []
+	_leyLines = {
+		current: null,
+		history: [],
+	}
+	_buffs = {}
+	_group = null
 
 	constructor(...args) {
 		super(...args)
-		this.addHook('cast', {
-			by: 'player',
-			abilityId: ACTIONS.LEY_LINES.id,
-		}, this._onCastLeyLines)
+
 		this.addHook('applybuff', {
 			by: 'player',
 			abilityId: STATUSES.CIRCLE_OF_POWER.id,
 		}, this._onApplyCircleOfPower)
+		this.addHook('applybuff', {
+			by: 'player',
+			abilityId: STATUSES.LEY_LINES.id,
+		}, this._onApplyLeyLines)
 		this.addHook('removebuff', {
 			by: 'player',
 			abilityId: STATUSES.CIRCLE_OF_POWER.id,
 		}, this._onRemoveCircleOfPower)
+		this.addHook('removebuff', {
+			by: 'player',
+			abilityId: STATUSES.LEY_LINES.id,
+		}, this._onRemoveLeyLines)
 		this.addHook('complete', this._onComplete)
+
+		this._group = new Group({
```

Comment:
I was questioning adding the new group at first, but looking at the example log, I can see how it'd be useful.

Perhaps down the line the group can be generalised or something, but it does the job nicely for now.